infrastructure/tooling/src/smoketest/checks/role.js

```diff
@@ -28,12 +28,16 @@ exports.tasks.push({
     };
     assert.deepEqual(expandedRole.scopes, expectedScopes.scopes);
     const query = {};
-    const anHourAgo = Date.now() - (1000*60*60);
+    const date = new Date();
+    const currentHour = date.getHours();
     while (1) {
       const res = await auth.listRoles2();
-      for(let i=0;i<res.roles.length;i++){
-        if(res.roles[i].roleId.includes('project:taskcluster:smoketest:') && res.roles[i].lastModified < anHourAgo){
-          await auth.deleteRole(res.roles[i].roleId);
+      for(let index in res.roles){
```

Comment:
Even better: `for (let role of res.roles)` -- no need for indexing at all!  The construct you've used actually iterates over the properties of `res.roles`, which for an array happen to be the array indexes, but *as strings* and *not in numeric order*.

Reply:
This is so much better. Thank you very much!